src/App.tsx

```diff
@@ -1,15 +1,57 @@
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropdownMenu';
+import { PersonData } from './types/PersonData';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectPerson, setSelectPerson] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPersonData, setSelectedPersonData] =
+    useState<PersonData | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name
+        .toLowerCase()
+        .trim()
+        .includes(appliedQuery.toLowerCase().trim()),
+    );
+  }, [appliedQuery]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setSelectPerson(value);
+
+    if (!value.trim()) {
+      setAppliedQuery('');
+      setSelectedPersonData(null);
+    } else {
+      applyQuery(value.trim());
+    }
+  };
+
+  useEffect(() => {
+    if (selectPerson.trim() && filteredPeople.length) {
```

Comment:
This is a code quality issue. You are directly interacting with the DOM by checking if `selectPerson.trim()` and `filteredPeople.length` are truthy. Instead, you should use React's state and props to handle this logic.